src/GoTrueClient.ts

```diff
@@ -1194,32 +1216,17 @@ export default class GoTrueClient {
    * Recovers the session from LocalStorage and refreshes
    * Note: this method is async to accommodate for AsyncStorage e.g. in React native.
    */
-  private async _recoverAndRefresh() {
+  private async _recoverAndRefresh(isInitial: boolean) {
     try {
-      const currentSession = await getItemAsync(this.storage, this.storageKey)
-      if (!this._isValidSession(currentSession)) {
-        if (currentSession !== null) {
-          await this._removeSession()
-        }
-
-        return
+      const {
+        data: { session: currentSession },
+        error,
+      } = await this._getSession()
+      if (error) {
+        throw error
       }
 
-      const timeNow = Math.round(Date.now() / 1000)
-
-      if ((currentSession.expires_at ?? Infinity) < timeNow + EXPIRY_MARGIN) {
-        if (this.autoRefreshToken && currentSession.refresh_token) {
-          const { error } = await this._callRefreshToken(currentSession.refresh_token)
-
-          if (error) {
-            console.log(error.message)
-            await this._removeSession()
-          }
-        }
-      } else {
-        if (this.persistSession) {
-          await this._saveSession(currentSession)
-        }
```

Comment:
No need to have 2 different versions that do refreshing when `getSession` does it all. It uses `_getSession` because `_recoverAndRefresh` is part of the `_initialize` flow and it means that `getSession` would never complete.

Reply:
Ah okay, so to expand -  if we called `getSession` instead of `_getSession()`  this would fail as `getSession` would call [intializePromise](https://github.com/supabase/gotrue-js/pull/684/files#diff-3522461172efd6058d6b8da62fc2d30d8b524d2b64894ea2c67218c52f7fdff5R685) which would  call `_initialize()` which would then call `_recoverAndRefresh` leading to an infinite loop?